library/core/functions.general.php

```diff
@@ -3565,6 +3565,26 @@ function trustedDomains() {
     }
 }
 
+if (!function_exists('isTrustedUrl')) {
```

Comment:
We already have a similar function!
See https://github.com/vanilla/vanilla/issues/5348